### PR TITLE
Add full documentation

### PR DESCRIPTION
**Pull request by klzns**
#### What does this PR do? \*

Add docs to render-runtime! 

#### How to test it? \*

Reading it: https://github.com/vtex-apps/render-runtime/tree/docs

#### Describe alternatives you've considered, if any. \*

We should add some JSDoc too, but we can do that later.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,143 +1,580 @@
...
 
-```javascript
-<NoSSR onSSR={<div>Loading...</div>}>
-  <DomRelatedComponent/>
-</NoSSR>
+<!-- /code_chunk_output -->
+
+## useRuntime
+
+A React hook that provides some functions and variables that are useful when creating components.
```
Comment (kaisermann) [inline docs/README.md:22]:
```suggestion
A React hook that provides runtime contextual variables and methods to help creating components.
```

File: docs/README.md [hunk 2]
```diff
@@ -1,143 +1,582 @@
...
+  "mobile": false,
+  "tablet": false,
+  "phone": false,
+  "unknown": false,
+}
+```
+
+### `history`
+
+A `history` object reexported from history package.
```
Comment (kaisermann) [inline docs/README.md:255]:
```suggestion
A `history` object reexported from the `history` package. 
```

File: docs/README.md [hunk 3]
```diff
@@ -1,143 +1,582 @@
...
 
 ### `navigate`
-This method is the most powerful solution for navigation and can be used inside a React Component's lifecycle. It may be injected with the HOC `withRuntimeContext` or, preferably, with the `useRuntime` hook.
-```javascript
-import { useRuntime, withRuntimeContext } from 'vtex.render-runtime'
-...
-const MyComponent = () => {
-  const { navigate }  = useRuntime()
+
+A _function_ that should be use to make a client-side navigation.
```
Comment (kaisermann) [inline docs/README.md:275]:
```suggestion
A _function_ that must be used to make a client-side navigation.
```

File: docs/README.md [hunk 4]
```diff
@@ -1,143 +1,582 @@
...
+    top: number
+  }
+  skipSetPath?: boolean
+  to?: string
 }
+```
+
+### `page`
+
+A _string_ value of the current page.
```
Comment (kaisermann) [inline docs/README.md:321]: Representing what? The URL?

File: docs/README.md [hunk 5]
```diff
@@ -1,143 +1,582 @@
...
-| fetchPage     | `boolean`   | `true`  | If `false`, won't fetch navigation assets in `pages-graphql`
-| page     | `string`  | --  | Name of the page that will be redirected to. Maps to a `blocks.json` block. Example: `'store.product'`
-| to     | `string`    |  --  | Alternatively to `page`, you can pass the whole URL directly instead of the page name (Useful for the `search-result`). Example: `/shirt/p?skuId=1`
-| params | `object`      |   `{}`  | Map of _parameters_ names in the path for the page and the values that should replace them. Example: `{slug: 'shirt'}`
-| query | `string`  | `''`   | String representation of the query params that will be appended to the path. Example: `skuId=231`.
-| scrollOptions | `RenderScrollOptions` | -- | After the navigation, if the page should be scrolled to a specific position, or should stay still (use `false`)
-| replace | `boolean` | `undefined` | If it should call the replace function to navigate or not
-| preventRemount | `boolean` | `false` | If `true`, only the URL will change, but not the components :exclamation: **Use with caution!**
+A _boolean_ value whether is in a production workspace or not.
```
Comment (kaisermann) [inline docs/README.md:342]:
```suggestion
A _boolean_ value representing whether the app is in a production workspace or not.
```

File: docs/README.md [hunk 6]
```diff
@@ -1,143 +1,582 @@
...
+    </>
+  )
+}
+
+export default MyComponent
+```
+
+## Link
+
+A React component that renders an `a` HTML element that, when clicked, navigates the user to the provided route. It has a similar API with the `navigate` method.
```
Comment (kaisermann) [inline docs/README.md:536]:
```suggestion
A React component that renders an `a` HTML element that, when clicked, navigates the user to the provided route. It has a similar API to the `navigate` method.
```

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 4 above)
Comment (klzns) [reply docs/README.md]: Good point!

File: docs/README.md [hunk 7]
```diff
@@ -1,143 +1,581 @@
 # Render Runtime
 
-This app handles runtime execution of React apps in the VTEX IO Platform.
+This app handles runtime execution of React apps in the VTEX IO Platform and exports some helpful React components, React hooks and variable.
```
Comment (kaisermann) [inline docs/README.md:3]:
```suggestion
This app handles runtime execution of React apps in the VTEX IO Platform and exports some helpful React components hooks and variables.
```

File: docs/README.md [hunk 8]
```diff
@@ -1,143 +1,581 @@
...
 
-## Exported Components
+> **Tip:** run `vtex setup --typings` to add vtex.render-runtime types in your app. This will provide autocomplete features in VSCode.
```
Comment (kaisermann) [inline docs/README.md:5]:
```suggestion
> **Tip:** run `vtex setup --typings` to add `vtex.render-runtime` types in your app. IDEs can use them to provide autocomplete for variables and methods.
```

**Review by kaisermann — APPROVED** [review]

File: docs/README.md [hunk 9]
```diff
@@ -1,143 +1,581 @@
...
+
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { hints } = useRuntime()
+
+  return <div>Hello</div>
```
Comment (victorhmp) [inline docs/README.md:246]:
```suggestion
  return <div>This {hints.desktop ? 'is' : 'is not'} a desktop</div>
```

File: docs/README.md [hunk 10]
```diff
@@ -1,143 +1,581 @@
...
+
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { page } = useRuntime()
+
+  return <div>Hello</div>
```
Comment (victorhmp) [inline docs/README.md:333]:
```suggestion
  return <div>This is the current page id: {page}</div>
```

File: docs/README.md [hunk 11]
```diff
@@ -1,143 +1,581 @@
...
+
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { production } = useRuntime()
+
+  return <div>Hello</div>
```
Comment (victorhmp) [inline docs/README.md:354]:
```suggestion
  return <div>This {production ? 'is' : 'is not'} a production workspace</div>
```

File: docs/README.md [hunk 12]
```diff
@@ -1,143 +1,581 @@
...
+
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { renderMajor } = useRuntime()
+
+  return <div>Hello</div>
```
Comment (victorhmp) [inline docs/README.md:396]:
```suggestion
  return <div>This page is rendered using vtex.render-runtime@{renderMajor}.x</div>
```

File: docs/README.md [hunk 13]
```diff
@@ -1,143 +1,581 @@
...
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { rootPath } = useRuntime()
+
+  return <div>Hello</div>
+}
```
Comment (victorhmp) [inline docs/README.md:418]: Could not think of a good example here, but we should also use the `rootPath` variable in some way here.

File: docs/README.md [hunk 14]
```diff
@@ -1,143 +1,581 @@
...
+
+**Usage:**
+```tsx
+import React from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+
+function MyComponent() {
+  const { workspace } = useRuntime()
+
+  return <div>Hello</div>
```
Comment (victorhmp) [inline docs/README.md:460]:
```suggestion
  return <div>This is the {workspace} workspace</div>
```

File: docs/README.md [hunk 15]
```diff
@@ -1,143 +1,581 @@
...
+}
+
+export default MyComponent
+```
+
+## Block (alias ExtensionPoint)
+
+A React component that is used to create Store Framework blocks that expects a specific block id.
```
Comment (victorhmp) [inline docs/README.md:468]: We could add a disclaimer here, telling people that they should use `Block` or `ExtensionPoint` if they expect to receive blocks via the `blocks` array.

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Review by victorhmp — APPROVED** [review]

**Comment by carolinamenezes** [comment]
@klzns @victorhmp atualizei o readme. se vocês não tiverem gostado dessa estrutura que coloquei, é só me falar que a gente volta hehe

**Pull request merged**